addons/account_payment/controllers/portal.py

```diff
@@ -20,7 +20,5 @@ def _invoice_get_page_view_values(self, invoice, access_token, **kwargs):
                                                                     ])
             values['existing_token'] = token_count > 0
         values.update(payment_inputs)
-        # if the current user is connected we set partner_id to his partner otherwise we set it as the invoice partner
-        # we do this to force the creation of payment tokens to the correct partner and avoid token linked to the public user
-        values['partner_id'] = invoice.partner_id if is_public_user else request.env.user.partner_id,
+        values['partner_id'] = invoice.partner_id  # should always be the billing partner
```

Comment:
This will not work properly, check the commit https://github.com/odoo/odoo/commit/bae7cd9c4695b264a3e8bb55054ac60877101b93#diff-d51ddfeb4801251efa6496488a55ba14R74 that added this line of code